Firestore/core/test/unit/credentials/firebase_auth_credentials_provider_test.mm

```diff
@@ -153,6 +153,29 @@ - (void)getTokenForcingRefresh:(BOOL)forceRefresh
   });
 }
 
+// Catch race conditions in FirebaseAuthCredentialsProvider::GetToken() when
+// run under thread sanitizer.
+// See https://github.com/firebase/firebase-ios-sdk/issues/10393
+TEST(FirebaseAuthCredentialsProviderTest, GetTokenCalledByAnotherThread) {
+  FIRApp* app = testutil::AppForUnitTesting();
+  FSTAuthFake* auth = [[FSTAuthFake alloc] initWithToken:@"token for fake uid"
+                                                     uid:@"fake uid"];
+  FirebaseAuthCredentialsProvider credentials_provider(app, auth);
+  credentials_provider.InvalidateToken();
```

Comment:
Could you please tell me why we need to call InvalidateToken()?

Reply:
I honestly don't know. I just copied a test from above and tweaked it. If it blocks your code review, I can investigate further though.

Reply:
Thank you!

Reply:
Actually... you're right. I had copied the code above, which is specifically testing InvalidateToken(). It probably isn't relevant for this test. I'll remove it.